R/pkg/R/DataFrame.R

```diff
@@ -1384,7 +1384,7 @@ setMethod("saveAsTable",
                                     "org.apache.spark.sql.parquet")
             }
             allModes <- c("append", "overwrite", "error", "ignore")
-            if (!(mode %in% allModes)) {
+            if (! (mode %in% allModes)) {
```

Comment:
I checkout the source code of the R interpreter, the most common style of using ! is:
  !<var>,  !<function call> and !(<expression>)
while there is still in-consistency of this style in the R source code like:
  !<space><expression>, ! <function call>, ...

I prefer to the former style. Take the following as example:
  ! dots.or.missing(e) && length(e) == 3
The real meaning of the expression is (! dots.or.missing(e)) && length(e) == 3, but seems like
  ! (dots.or.missing(e) && length(e) == 3)
So the following looks better:
!dots.or.missing(e) && length(e) == 3